pkg/apis/sessions/session_state.go

```diff
@@ -141,7 +198,7 @@ func legacyDecodeSessionStatePlain(v string) (*SessionState, error) {
 
 // legacyDecodeSessionState attempts to decode the session state string
 // generated by v3.1.0 or older
-func legacyDecodeSessionState(v string, c *encryption.Cipher) (*SessionState, error) {
+func legacyV3DecodeSessionState(v string, c *encryption.Cipher) (*SessionState, error) {
```

Comment:
Perhaps it's time to drop this? It's been a while, people should have upgraded by now. I might raise a separate PR for that

Reply:
I left a note in this PR: https://github.com/oauth2-proxy/oauth2-proxy/pull/524#issuecomment-625448814

The moment we trigger a hard shift to SHA256 signed cookies instead of SHA1, we can ditch all Legacy code, since the old SHA1 cookies won't validate.

Is that a good time to drop all these legacy areas?